src/iso15118/d20/state/dc_charge_loop.cpp

```diff
@@ -169,21 +167,22 @@ handle_request(const message_20::DC_ChargeLoopRequest& req, const d20::Session&
 
     if (std::holds_alternative<Scheduled_DC_Req>(req.control_mode)) {
 
-        if (session.get_selected_energy_service() != message_20::ServiceCategory::DC) {
+        if (session.get_selected_control_mode() != message_20::ControlMode::Scheduled or
+            session.get_selected_energy_service() != message_20::ServiceCategory::DC) {
```

Comment:
Not sure here, but these are places in the code where a comment might be helpful to explain why in this case it is not allowed ...

Reply:
I added comments to explain these cases